### DISCUSSION TITLE
Context menu with near by words

### DISCUSSION THREAD

**Discussion by rohan-gosavi** [stephanrauh/ngx-extended-pdf-viewer#2542] [Q&A]
How can i build a context menu with few nearby words?
I want to build custom cards with this info and when user clicks on the card page will scroll to that section of the pdf.

**Comment by stephanrauh** [comment]
I'm afraid you have to give us some context (no pun intended). I know the meaning of "context menu with a few nearby words" is obvious to you, but even so, I don't understand your question. I can start guessing, but more likely than not, I'll guess wrong.

Would you mind to elaborate your question?

**Comment by rohan-gosavi** [comment]
Hi @stephanrauh ,

Please find details:

In below screenshot i am searching for '**work**' so can i get the content of that word like 4-5 words before and after of the work e.g. '**The author has made an online version of this work  available under a Creative**' as a card or some clickable element.
When i click on these cards then i want to scroll to that area of the pdf.

![image](https://github.com/user-attachments/assets/0fbc4f4a-5b1a-4b33-8678-76bd9b6fd9c0)

**Reply by rohan-gosavi** [reply]
Hi @stephanrauh,

Is there anything i can do to achieve this?

**Reply by stephanrauh** [reply]
I guess you can achieve this by manipulating the text layer and by listening to an event like `(updateFindMatchesCount)` or `(updateFindState)`. These events are - probably - raised after marking the text in the text layer. Your event listener can analyze the text layer - which is a `<div>` containing many `<span>` tags - and search for the `<span>` tag bearing the `selected` class. The parent tag usually contains a couple of words around the highlighted text. Add a `onClick` listener to that parent tag, and you're reached your goal.
